lib/Reader/reader.cpp

```diff
@@ -8625,3 +8662,166 @@ void ReaderBase::resolveToken(mdToken Token, CorInfoTokenKind TokenType,
   return resolveToken(Token, getCurrentContext(), getCurrentModuleHandle(),
                       TokenType, ResolvedToken);
 }
+
+#pragma region SIMD_INTRINSICS
+
+//===----------------------------------------------------------------------===//
+//
+// SIMD Intrinsics
+//
+//===----------------------------------------------------------------------===//
+
+IRNode *ReaderBase::generateSIMDBinOp(ReaderSIMDIntrinsic OperationCode) {
+  IRNode *Arg2 = ReaderOperandStack->pop();
+  IRNode *Arg1 = ReaderOperandStack->pop();
+  if (checkVectorType(Arg1) && checkVectorType(Arg2)) {
+
+    IRNode *Vector1 = Arg1;
+    IRNode *Vector2 = Arg2;
+    
+    IRNode *opResult = 0;
+    switch (OperationCode) {
+    case ADD:
+      opResult = vectorAdd(Vector1, Vector2);
+      break;
+    case SUB:
+      opResult = vectorSub(Vector1, Vector2);
+      break;
+    case MUL:
+      opResult = vectorMul(Vector1, Vector2);
+      break;
+    case DIV:
+      opResult = vectorDiv(Vector1, Vector2);
+      break;  
+    default:
+      assert(UNREACHED);
+    }
+    if (opResult) {
+      return opResult;
+    }
+  }
+
+  ReaderOperandStack->push(Arg1);
+  ReaderOperandStack->push(Arg2);
+  return 0;
+}
+
+IRNode *ReaderBase::generateSIMDUnOp(ReaderSIMDIntrinsic OperationCode) {
+  IRNode *Arg = ReaderOperandStack->pop();
+  IRNode *Vector = Arg;
+  if (Vector) {
+    IRNode *opResult = 0;
+    switch (OperationCode) {
+    case ABS:
+      opResult = vectorAbs(Vector);
+      break;
+    case SQRT:
+      opResult = vectorSqrt(Vector);
+      break;
+    default:
+      assert(UNREACHED);
+    }
+    return opResult;
+  }
+  ReaderOperandStack->push(Arg);
+
+  return 0;
+}
+
+
+IRNode *ReaderBase::generateSIMDIntrinsicCall(CORINFO_CLASS_HANDLE Class,
+  CORINFO_METHOD_HANDLE Method, CORINFO_SIG_INFO *SigInfo, ReaderBaseNS::CallOpcode Opcode) {
+
+  const char *ModuleName;
+  const char *MethodName = GetMethodName(Method, &ModuleName, JitInfo);
+  if (!strcmp(MethodName, "get_IsHardwareAccelerated")) { // Special case.
+    return generateIsHardwareAccelerated(Class);
+  } 
+  
+  uint32_t ArgsCount = SigInfo->numArgs;
+  assert(ArgsCount <= ReaderOperandStack->size());
+  if (SigInfo->hasThis()) {
+    ArgsCount += 1;
```

Comment:
Given that the only user of this value is `generateSIMDCtor` and that method always uses `ArgsCount - 1`, I think we can remove this adjustment.